### PR TITLE
[v10] Merge `release-10.0` into `main`

### PR DESCRIPTION
**Pull request by andrewheard**
Merge the `release-10.0` branch into `main` in preparation for the GDT 10.0 release.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -28,19 +28,12 @@ let package = Package(
   ],
   dependencies: [
     .package(
-      name: "nanopb",
       url: "https://github.com/firebase/nanopb.git",
       "2.30908.0" ..< "2.30911.0"
```
Comment (ncooke3) [inline Package.swift:32]: This may be out of sync with podspec dependency:
```ruby
s.dependency 'nanopb', '~> 3.30910.0'
```

**Review by ncooke3 — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (ncooke3) [reply Package.swift:32]: firebase-ios-sdk [specifies](https://github.com/firebase/firebase-ios-sdk/blob/856e201e5dbe0e7096a3f7a0a409deae0d79d8af/Package.swift#L151-L154):
```swift
    .package(
      url: "https://github.com/firebase/nanopb.git",
      "2.30909.0" ..< "2.30911.0"
    ),
```

Comment (ncooke3) [reply Package.swift:32]: (Doesn't block this PR) But I think we should bump the Package.swift nanopb lower bound to `2.30910.0`, and maybe reconsider the ~> operator in the podspec.

Comment (paulb777) [reply Package.swift:32]: Any reason not to go ahead and release 3.30910.0 for both CocoaPods and SPM? Then we can uniformly update nanopb's clients for Firebase 11.

Comment (ncooke3) [reply Package.swift:32]: Resolving. IIUC, `'nanopb', '~> 3.30910.0'` is the same as 'nanopb', '>= 3.30910.0', '< 3.30911.0'

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**